api/v1alpha2/hierarchy_types.go

```diff
@@ -147,6 +161,18 @@ type HierarchyConfigurationList struct {
 	Items           []HierarchyConfiguration `json:"items"`
 }
 
+// MetaKVP represents a label or annotation
+type MetaKVP struct {
+	// Key is the name of the label or annotation. It must conform to the normal rules for Kubernetes
+	// label/annotation keys.
+	Key string `json:"name"`
```

Comment:
Is `json:"name"` correct, or should it be `json:"key"`?

(the field is `Key` and this error message says `key`: `log.Info("Illegal managed label", "key", kvp.Key)`)

Reply:
Crap, good catch! I forgot to change that tag when I changed the field name from "Name" to "Key." Fixed.

I guess this shows why we need to add end-to-end tests, not just integ tests :)